src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs

```diff
@@ -8,11 +8,39 @@
 using System.ComponentModel;
 using System.Diagnostics;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
+    public static class SeparatedSyntaxList
+    {
+        public static SeparatedSyntaxList<TNode> Create<TNode>(ReadOnlySpan<TNode> nodes) where TNode : SyntaxNode
+        {
+            if (nodes.Length == 0)
+                return default;
+
+            if (nodes.Length == 1)
+                return new SeparatedSyntaxList<TNode>(new SyntaxNodeOrTokenList(nodes[0], index: 0));
+
+            var builder = new CodeAnalysis.Syntax.SeparatedSyntaxListBuilder<TNode>(nodes.Length);
+
+            builder.Add(nodes[0]);
+
+            var separator = nodes[0].Green.CreateSeparator(nodes[0]);
+
+            for (int i = 1, n = nodes.Length; i < n; i++)
+            {
+                builder.AddSeparator(separator);
```

Comment:
How did you decide to add separators vs. just inserting the provided nodes?

Reply:
this is the overload that takes in nodes-only, but is producing a separated list.  the semantics are that we must separate the nodes with the language separator.  Note: this is how teh existing SyntaxFactory.SeparatedList helper worked (though it could hardcode in directly the comma token at the lang specific level).  At this level, we need the virtual call to get teh separator as it will be different from vb to c#.